src/routes/tools/local-climate-change-snapshot/explore.svelte

```diff
@@ -173,19 +187,41 @@
   async function initApp() {
     const { query } = $page;
     // Get initial configuration (from default or from url)
-    const { lat, lng, boundary, indicator, imperial } = getInitialConfig(
-      query,
-      DEFAULT_INITIAL_CONFIG
-    );
+    const {
+      lat,
+      lng,
+      boundary: boundaryType,
+      indicator,
+      imperial,
+    } = getInitialConfig(query, DEFAULT_INITIAL_CONFIG);
     // Set intial values for stores
+
+    if (!$location || !$boundary) {
+      await setLocationStoreOnLoad(+lat, +lng, boundaryType);
+    }
     categoryListStore.set(categories);
     indicatorListStore.set(indicators);
     indicatorStore.set(indicators.find((d) => d.id === indicator));
     unitsStore.set({ imperial });
-    const loc = await setInitialLocation(+lng, +lat, boundary);
+  }
+
+  async function setLocationStoreOnLoad(lat, lng, boundaryType) {
+    let loc = DEFAULT_LOCATION;
+    if (
+      isValidNumber(lat) &&
```

Comment:
maybe could wrap this in an `isValidLocationParams` function:

```js
function isValidLocationParams(lng, lat, boundaryType) {
  return isValidNumber(lng) && isValidNumber(lat) && VALID_BOUNDARY_TYPES.has(boundaryType)
}
```
so that it can be shared between the index and explore pages.